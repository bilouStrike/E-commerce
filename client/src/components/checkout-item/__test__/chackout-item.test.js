import { shallow } from 'enzyme';
import React from 'react';
import { CheckoutItem } from '../checkout-item.component';
import toJson from 'enzyme-to-json';

describe("Checkout component:", () => {
 
    const MockcartItem = { 
        name: 'Product1',
        imageUrl: 'image url',
        price: 50,
        quantity: 1
    };

    const mockRemoveItem = jest.fn();
    const wrapper = toJson(shallow(<CheckoutItem cartItem={MockcartItem} removeItem={mockRemoveItem}/>));

    it('expect checkout component to do onot changing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to invok removeItem on click in removeItem Button', () => {
        const wrapper2 = shallow(<CheckoutItem cartItem={MockcartItem} removeItem={mockRemoveItem}/>);
        wrapper2.find('[id="removeItem"]').simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
    });

});