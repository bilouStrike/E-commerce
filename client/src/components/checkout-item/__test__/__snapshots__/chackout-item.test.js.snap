// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component: expect checkout component to do onot changing 1`] = `
<div
  className="checkout-item"
>
  <div
    className="image-container"
  >
    <img
      alt="item"
      src="image url"
    />
  </div>
  <span
    className="name"
  >
    Product1
  </span>
  <span
    className="quantity"
  >
    1
  </span>
  <span
    className="price"
  >
    50
  </span>
  <div
    className="remove-button"
    id="removeItem"
    onClick={[Function]}
  >
    âœ•
  </div>
</div>
`;
